components/forms/Login.tsx

```diff
@@ -0,0 +1,140 @@
+import { Dispatch, SetStateAction } from 'react';
+import styles from '@/styles/Login.module.css';
+import {
+  createSessionSchema,
+  CreateSessionInput,
+} from '@/types/client/session';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useRouter } from 'next/router';
+import { useSupabaseClient } from '@supabase/auth-helpers-react';
+import { Database } from '@/types/database.supabase';
+
+export default function Login({
+  setServerError,
+}: {
+  setServerError: Dispatch<SetStateAction<string | null>>;
+}) {
+  const router = useRouter();
+  const supabase = useSupabaseClient<Database>();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<CreateSessionInput>({
+    resolver: zodResolver(createSessionSchema),
+  });
+
+  const onSubmit = async (formData: CreateSessionInput) => {
+    const { data, error } = await supabase.auth.signInWithPassword({
+      email: formData.email,
+      password: formData.password,
+    });
+
+    if (error) {
+      setServerError(error.message);
+      setTimeout(() => {
+        setServerError(null);
+      }, 7000);
+    }
+    if (data && !error) {
+      router.push('/');
+    }
+  };
+  return (
+    <form className="flex flex-col" onSubmit={handleSubmit(onSubmit)}>
+      <div className="relative mt-7">
+        <div className={`${errors.email ? styles.iconError : styles.icon} `}>
+          <svg
+            width="18"
+            height="18"
+            viewBox="0 0 34 28"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path
+              d="M30.3333 0.666626H3.66667C1.83334 0.666626 0.350003 2.16663 0.350003 3.99996L0.333336 24C0.333336 25.8333 1.83334 27.3333 3.66667 27.3333H30.3333C32.1667 27.3333 33.6667 25.8333 33.6667 24V3.99996C33.6667 2.16663 32.1667 0.666626 30.3333 0.666626ZM30.3333 7.33329L17 15.6666L3.66667 7.33329V3.99996L17 12.3333L30.3333 3.99996V7.33329Z"
+              fill="white"
+            />
+          </svg>
+        </div>
+        <input
+          type="email"
+          className={`form-control
+                w-full
+                py-2 
+                pl-6
+                          self-start
+                          text-base
+                          font-normal
+                          placeholder:text-white
+                          placeholder:opacity-60     
+                          border-2 border-solid border-blueGrey-600
+                          rounded-2xl
+                          transition
+                          ease-in-out
+                          focus:outline-frost-50
+                          m-0 focus:outline-none  bg-inherit flex-1  ${styles.input}`}
```

Comment:
Firstly, indentation/formatting on this is inconsistent

Secondly, I noticed we're using this same block multiple times, is there any reason we're not using a const for this?

We could easily just define this in styles and apply it like that to preserve readability and just general cleanliness of the file